pkg/kubelet/server/stats/types.go

```diff
@@ -147,3 +151,37 @@ type FsStats struct {
 	// e.g. For ContainerStats.Rootfs this is the bytes used by the container rootfs on the filesystem.
 	UsedBytes *resource.Quantity `json:"usedBytes,omitempty"`
 }
+
+type UserDefinedMetricType string
+
+const (
+	// Instantaneous value. May increase or decrease.
+	MetricGauge UserDefinedMetricType = "gauge"
+
+	// A counter-like value that is only expected to increase.
+	MetricCumulative = "cumulative"
+
+	// Rate over a time period.
+	MetricDelta = "delta"
+)
+
+type UserDefinedMetricValue struct {
+	// Time at which the metric was queried
+	Timestamp time.Time `json:"timestamp"`
+
+	// The value of the metric at this point.
+	Value *resource.Quantity `json:"value,omitempty"`
```

Comment:
Does it make sense to have a "MetricValue" without a value? I.e. should this be required?